### PR TITLE
Introduce 1Inch HTTP Client

### PR DESCRIPTION
**Pull request by nlordell**
This PR introduces an HTTP client to the 1Inch API. This client will eventually be used by a "1Inch solver" that would be able to match single EOA orders to a 1Inch swap.

The HTTP client is currently incomplete and does not:
- Include all query parameters
- Parse the response JSON in its entirety

Although, it is enough for what we need for a 1Inch solver for now.

### Test Plan

Added new unit tests. To do a real HTTP request:
```
$ cargo test -- --ignored --nocapture oneinch_swap
...
running 1 test
Swap {
    ...
}
test oneinch_solver::api::tests::oneinch_swap ... ok
...
```

### CODE REVIEW

File: solver/src/oneinch_solver/api.rs [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+/// A 1Inch API quote query parameters.
+///
+/// These parameters are currently incomplete, and missing parameters can be
+/// added incrementally as needed.
+pub struct SwapQuery {
```
Comment (fleupold) [inline solver/src/oneinch_solver/api.rs:55]: Is it possible to model buy orders?

**Review by fleupold — APPROVED** [review]
Nice!

File: solver/src/oneinch_solver/api.rs (hunk 1 above)
Comment (nlordell) [reply solver/src/oneinch_solver/api.rs:55]: Good question...

Comment (nlordell) [reply solver/src/oneinch_solver/api.rs:55]: According to https://docs.1inch.io/api/quote-swap I don't see any obvious way of doing it :disappointed:

**Review by vkgnosis — APPROVED** [review]

**Pull request merged**